conversion.py

```diff
@@ -58,14 +58,30 @@ def convert(env,env_folder,cfg):
     with open(Path(env_folder) / env_name / 'problem.yaml', 'w') as outfile:
         yaml.dump(new_format, outfile)   
 
+def extract_results(env_file, models, ma_starts, ma_segs):
+    # get robot types
+    agent_types=[]
+    with open(env_file, "r") as file:
+        data = yaml.load(file, Loader=yaml.Loader)
+        robots = data['robots'] 
+    for i in range(len(robots)):
+        agent_types.append(robots[i]["type"])   
 
-
-# def main():
-#     parser = argparse.ArgumentParser(description='Convert our data format to s2s format')
-#     parser.add_argument('environment_yaml', help="environment.yaml file") 
-#     args = parser.parse_args()
-#     env_name = Path(args.environment_yaml).stem
-#     convert(args.environment_yaml, env_name)
-
-# if __name__ == "__main__":
-#     main()
\ No newline at end of file
+    agent_num = len(models)
+    paths = []
+    for idx in range(agent_num):
+        path = []
+        segs = ma_segs[idx]
+        start_state = ma_starts[idx]
+        if(agent_types[idx] == "unicycle_first_order_0"):
+            q0 = start_state + [0] # adding theta manually
```

Comment:
we actually sometimes start from non-zero theta's even in the existing swap2 example. Is there a way to read it from the problem description rather than hard-coding it?

Reply:
Yes, I can do it. It should be possible, it was their original implementation assumption when control were applied to the trajectory.